index.js

```diff
@@ -443,58 +445,84 @@ class HtmlWebpackInlineSVGPlugin {
+     * @param {Object{}} resolve - Consumer's Promise resolve hook
      * @returns {Promise}
      *
      */
-    processOutputHtml (html, inlineImage) {
+    optimizeSvg ({ html, inlineImage, data, resolve }) {
+        const configObj = Object.assign(svgoDefaultConfig, this.userConfig)
 
-        return new Promise((resolve, reject) => {
+        const config = {}
 
-            const svgSrc = this.getImagesSrc(inlineImage)
 
+        // pass all objects to the config.plugins array
 
-            // if the image isn't valid resolve
+        config.plugins = _.map(configObj, (value, key) => ({ [key]: value }));
 
-            if (!svgSrc) return resolve(html)
 
+        // create a new instance of SVGO
+        // passing it the merged config, to optimize the svg
 
-            // read in the svg
+        const svgo = new SVGO(config)
 
-            fs.readFile(path.resolve(this.outputPath, svgSrc), 'utf8', (err, data) => {
-
-                if (err) reject(err)
+        svgo.optimize(data)
+            .then((result) => {
 
-                const configObj = Object.assign(svgoDefaultConfig, this.userConfig)
+                const optimisedSVG = result.data
 
-                const config = {}
+                html = this.replaceImageWithSVG(html, inlineImage, optimisedSVG)
 
+                resolve(html)
 
-                // pass all objects to the config.plugins array
+            })
+            .catch((err) => console.log(chalk.red(err)))
+    }
 
-                config.plugins = _.map(configObj, (value, key) => ({ [key]: value }));
 
+    /**
+     * append the inlineImages SVG data to the output HTML and remove the original img by
+     * loading the SVG data from the filesystem or from an URL
+     * @param {string} html
+     * @param {Object} inlineImage - parse5 document
+     * @returns {Promise}
+     *
+     */
+    processOutputHtml (html, inlineImage) {
+        return new Promise((resolve, reject) => {
 
-                // create a new instance of SVGO
-                // passing it the merged config, to optimize the svg
+            const svgSrc = this.getImagesSrc(inlineImage)
 
-                const svgo = new SVGO(config)
+            // if the image isn't valid resolve
+            if (!svgSrc) return resolve(html)
 
-                svgo.optimize(data)
-                    .then((result) => {
+            // read in the svg
+            fs.readFile(path.resolve(this.outputPath, svgSrc), 'utf8', (err, data) => {
+                if (err) {
+                    // loading from the filesystem failed
+                    if (!this.allowFromUrl) {
+                        return reject(err)
+                    }
 
-                        const optimisedSVG = result.data
+                    axios.get(svgSrc)
+                        .then(({ status, data }) => {
+                            if (!status === 200) {
+                                resolve(html)
+                                return
```

Comment:
I'm not sure about this. But shouldn't this be `return reject(status)`?

Reply:
I run the tests with the changes I proposed and two assertions failed. The assertions are correct so that means my suggestion doesn't make sense. But at the same time don't understand the logic of the code since, if the URL path fails to gather the SVG content, should be the same result than when the SVG content fails to be read locally?

![image](https://user-images.githubusercontent.com/5444421/84696808-f74ee080-af44-11ea-8145-f73f1d0cb46b.png)
.

Reply:
Definitively there is some rogue bug happening here. With fresh changes from this current branch, if on the `index-allow-from-url.html` template I move the image with the valid URL to the last position the tests will fail:

![image](https://user-images.githubusercontent.com/5444421/84703512-44847f80-af50-11ea-939d-6e184f3cd467.png)

Reply:
You were right; there were bugs. I rethought the approach and now have submitted a better version.

However, when we're not able to retrieve an URL, we should not reject the promise mechanism but rather mark the image as `not retrievable` and keep iterating on the rest of the document.